### ISSUE TITLE
ability to generate a report via a modal form that includes the current map view

### ISSUE THREAD

**Issue by jywarren** [publiclab/leaflet-environmental-layers#32]
People often want to annotate a location with some text, for example if it's a pollution site. We should make something that opens a form you can add text to, and has a configurable function to submit it to a remote URL!

This could be partially broken out as a separate leaflet-form library at some point... 

Good reference here - https://publiclab.org/notes/warren/09-20-2017/compile-screenshots-and-urls-for-aerial-images-of-harvey-damages-in-a-spreadsheet-for-reporting

It could also include the layers that are being viewed during the report!

**Comment by sagarpreet-chadha** [comment]
Hi @jywarren 

Can you elaborate these points : 
1.) Form or something like that should open on map and then we add some text and automatically take latitude and longitude , AND the layers currently selected , right ?
2.) And which remote URL should we use ?
3.) And where to use this information collected ?

Thanks !

**Comment by jywarren** [comment]
I think we should think about a default URL where it could submit - maybe
to publiclab.org/post?title=_____&body=______&tags=_____ for starters? But
think about it as a general purpose form.

Does that make sense?

**Comment by sagarpreet-chadha** [comment]
Okay makes sense .

Can you help me break this issue into steps ?

**Comment by jywarren** [comment]
Yeah - perhaps the first thing would be to make a listener and a button,
where the button can be pressed to run a given function where:

function onFormOpen(parameters) {

and then run a second on:

function onFormSubmit(parameters) {

so that you can access parameters.location.latitude,
parameters.location.longitude,
and parameters.location.layers, maybe?

So before all that, maybe some general purpose function to return the
layers, although Leaflet may have a standard way to do this?

How's that for a start?

**Comment by sagarpreet-chadha** [comment]
Hi @jywarren ...should we write this code in plots2 after we integrate leaflet-environmental-layers ? or inside `<script>` tag in _index.html_ for now ?

**Comment by jywarren** [comment]
Let's start by writing it into leaflet-environmental-layers, because
perhaps other people will want to integrate such a function into their own
pages. We can provide template HTML to make this work, but make it flexible
enough to adapt to other websites too. What do you think?

**Comment by ananya** [comment]
I would like to work on this. Is the issue still open? Please guide me on how to proceed

**Comment by sagarpreet-chadha** [comment]
Yes please :)

**Comment by ananya** [comment]
This will be my first contribution to the project, please help me with how to start.
